app/payment-link-v2/amount/get-index.controller.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+// Custom dependencies
+const { response } = require('../../utils/response')
+const { getSessionVariable } = require('../../utils/cookie')
+const paths = require('../../paths')
+const replaceParamsInPath = require('../../utils/replace-params-in-path')
+const { NotFoundError } = require('../../errors')
+
+module.exports = (req, res, next) => {
+  const product = req.product
+  const data = {
+    productExternalId: product.external_id,
+    productName: product.name,
+  }
+
+  if (product.price) {
+    return next(new NotFoundError('Attempted to access amount page with a product that already has a price.'))
+  }
+
+  const sessionAmount = getSessionVariable(req, 'amount')
+
+  if (sessionAmount) {
+    data.backLinkHref = replaceParamsInPath(paths.paymentLinksV2.confirm, product.external_id)
+    data.productAmount = (sessionAmount / 100).toFixed(2)
+  } else if (product.reference_enabled) {
+    const sessionReferenceNumber = getSessionVariable(req, 'referenceNumber')
+    if (!sessionReferenceNumber) {
```

Comment:
Looks like if `product.reference_enabled` is true and there is a `sessionReferenceNumber` we won't be setting the `backLinkHref` to anything

Reply:
Good spot - we no longer check for the session variable `referenceNumber` - we now just set the back link to the `confirm` page in all instances when `product.reference_enabled=true`.
Also updated the tests as well.